base/compiler/tfuncs.jl

```diff
@@ -971,6 +971,44 @@ function _getfield_tfunc(@nospecialize(s00), @nospecialize(name), setfield::Bool
     return rewrap_unionall(R, s00)
 end
 
+function getfield_notundefined(@nospecialize(obj), @nospecialize(name))
+    if !isa(name, Const)
+        if !hasintersect(widenconst(name), Union{Int,Symbol})
+            # always throw doesn't account for undefined
```

Comment:
I don't understand this comment

Reply:
I think we don't need to taint `:consistent`-cy when a `getfield` call is determined to throw. If the thrown object (with a potentially undefined field) is caught later, we will taint it anyway when we see `:the_exception` expression.

Reply:
That is correct